### PR TITLE
clean up language error and fluffy comments

### PR DESCRIPTION
**Pull request by MikeLing**
This pr is about fixing language error and fluffy comments in this pr https://github.com/shogun-toolbox/shogun/pull/3608. For more information, see @karlnapf 's comments in  https://github.com/shogun-toolbox/shogun/pull/3608#pullrequestreview-22316735

### CODE REVIEW

File: src/shogun/multiclass/BruteKNNSolver.h [hunk 1]
```diff
@@ -16,8 +16,8 @@ namespace shogun
 {
 
 /**
- * The BruteKNNSolver class is a solver class which inherit from KNNSolver class. It will use brute way
- * to classify the object which mean compare the object against all training objects for each prediction.
+ * The BruteKNNSolver class is a solver class and it use brute way to classify the object,
+ * which means test points are compared to all training data for each prediction.
```
Comment (karlnapf) [inline src/shogun/multiclass/BruteKNNSolver.h]: can you pls use the one I wrote in the comment? 

"Test points are compared to all training data for each prediction."

File: src/shogun/multiclass/CoverTreeKNNSolver.h [hunk 2]
```diff
@@ -17,9 +17,9 @@ namespace shogun
 {
 
 /**
- * The CoverTreeKNNSolver class is a solver class which inherit from KNNSolver class. 
- * It use the cover tree to speed up the nearest neighbor search and you can find more
- * detail information on https://en.wikipedia.org/wiki/Cover_tree
+ * The CoverTreeKNNSolver class is a solver class.
```
Comment (karlnapf) [inline src/shogun/multiclass/CoverTreeKNNSolver.h]: No need for the first sentence (also in the other places)

File: src/shogun/multiclass/CoverTreeKNNSolver.h [hunk 3]
```diff
@@ -17,9 +17,9 @@ namespace shogun
 {
 
 /**
- * The CoverTreeKNNSolver class is a solver class which inherit from KNNSolver class. 
- * It use the cover tree to speed up the nearest neighbor search and you can find more
- * detail information on https://en.wikipedia.org/wiki/Cover_tree
+ * The CoverTreeKNNSolver class is a solver class. 
+ * It use the cover tree to speed up the nearest neighbor searching process.
```
Comment (karlnapf) [inline src/shogun/multiclass/CoverTreeKNNSolver.h]: Mistakes:
 * It uses (third person verbs have an s!)
 * It uses cover trees, or it uses the cover tree algorithm
 * .... to speed up the nearest neighbour search process (not searching process) ... Also it is not a search process, so rather write: to speed up the nearest neighbour computation

It uses cover

File: src/shogun/multiclass/CoverTreeKNNSolver.h [hunk 4]
```diff
@@ -17,9 +17,9 @@ namespace shogun
...
+ * For more information, see [https://en.wikipedia.org/wiki/Cover_tree]
```
Comment (karlnapf) [inline src/shogun/multiclass/CoverTreeKNNSolver.h]: why the square brackets around the link?

File: src/shogun/multiclass/KDTreeKNNSolver.h [hunk 5]
```diff
@@ -19,7 +19,7 @@ namespace shogun
  * The KDTREEKNNSolver class is a solver class which inherit from KNNSolver class.
  * It use a k-d tree (short for k-dimensional tree) to delivery the nearest neighbor searches
  * involving the multidimensional search key.
- * Check https://en.wikipedia.org/wiki/K-d_tree for more detail about K-D tree.
+ * For more information, see [https://en.wikipedia.org/wiki/K-d_tree]
```
Comment (karlnapf) [inline src/shogun/multiclass/KDTreeKNNSolver.h]: see above

File: src/shogun/multiclass/KDTreeKNNSolver.h [hunk 6]
```diff
@@ -19,7 +19,7 @@ namespace shogun
  * The KDTREEKNNSolver class is a solver class which inherit from KNNSolver class.
```
Comment (karlnapf) [inline src/shogun/multiclass/KDTreeKNNSolver.h]: This stuff needs change as well

File: src/shogun/multiclass/KNNSolver.h [hunk 7]
```diff
@@ -16,9 +16,7 @@
 namespace shogun
 {
 /**
- * The KNNSolver class is the virtual base class of BruteKNNSolcer, CoverTreeKNNSolver, KDTreeKNNsolver, LSHKNNSolver. 
- * The KNNSolver class will include most of Variables which been used all the other Solver and methods been used to choose 
- * the index of the most frequent class.
+ * The KNNSolver class is the virtual base class of all the other solvers.
```
Comment (karlnapf) [inline src/shogun/multiclass/KNNSolver.h]: "Virtual base class for all KNN solvers"

No need to put the class name in the description of a class, the name is obvious from the API

File: src/shogun/multiclass/LSHKNNSolver.h [hunk 8]
```diff
@@ -16,9 +16,9 @@ namespace shogun
 {
 
 /**
- * The LSHKNNSolver class is a solver class which inherit from KNNSolver class.
+ * The LSHKNNSolver class is a solver class.
```
Comment (karlnapf) [inline src/shogun/multiclass/LSHKNNSolver.h]: No need for this,

File: src/shogun/multiclass/LSHKNNSolver.h [hunk 9]
```diff
@@ -16,9 +16,9 @@ namespace shogun
...
  * It use LSH(short for Locality-sensitive hashing) to do the nearest neighbor search.
- * More detail information can be found in https://en.wikipedia.org/wiki/Locality-sensitive_hashing.
+ * More detailed information can be found in https://en.wikipedia.org/wiki/Locality-sensitive_hashing.
```
Comment (karlnapf) [inline src/shogun/multiclass/LSHKNNSolver.h]: Can you make this unified with above references

For more information, see ...

**Comment by MikeLing** [comment]
Hi @karlnapf, does it look better to you? :)

File: src/shogun/multiclass/BruteKNNSolver.h [hunk 10]
```diff
@@ -16,8 +16,7 @@ namespace shogun
 {
 
 /**
- * The BruteKNNSolver class is a solver class which inherit from KNNSolver class. It will use brute way
- * to classify the object which mean compare the object against all training objects for each prediction.
+ * Test points are compared to all training data for each prediction.
```
Comment (karlnapf) [inline src/shogun/multiclass/BruteKNNSolver.h]: Prepend: "Standard KNN solver. "

File: src/shogun/multiclass/CoverTreeKNNSolver.h [hunk 11]
```diff
@@ -17,9 +17,8 @@ namespace shogun
 {
 
 /**
- * The CoverTreeKNNSolver class is a solver class which inherit from KNNSolver class. 
- * It use the cover tree to speed up the nearest neighbor search and you can find more
- * detail information on https://en.wikipedia.org/wiki/Cover_tree
+ * It uses the cover trees to speed up the nearest neighbour computation.
```
Comment (karlnapf) [inline src/shogun/multiclass/CoverTreeKNNSolver.h]: Cover tree solver. It uses cover trees ...

File: src/shogun/multiclass/KDTreeKNNSolver.h [hunk 12]
```diff
@@ -16,10 +16,9 @@ namespace shogun
 {
 
 /**
- * The KDTREEKNNSolver class is a solver class which inherit from KNNSolver class.
- * It use a k-d tree (short for k-dimensional tree) to delivery the nearest neighbor searches
+ * It uses k-d tree (short for k-dimensional tree) to speed up the nearest neighbour computation.
```
Comment (karlnapf) [inline src/shogun/multiclass/KDTreeKNNSolver.h]: KD-tree solver. I uses kd-trees to speed up....

File: src/shogun/multiclass/LSHKNNSolver.h [hunk 13]
```diff
@@ -16,9 +16,8 @@ namespace shogun
 {
 
 /**
- * The LSHKNNSolver class is a solver class which inherit from KNNSolver class.
- * It use LSH(short for Locality-sensitive hashing) to do the nearest neighbor search.
- * More detail information can be found in https://en.wikipedia.org/wiki/Locality-sensitive_hashing.
+ * It uses LSH(short for Locality-sensitive hashing) to do the nearest neighbour computation.
```
Comment (karlnapf) [inline src/shogun/multiclass/LSHKNNSolver.h]: LSH solver.

Comment (karlnapf) [inline src/shogun/multiclass/LSHKNNSolver.h]: space between LSH and (

**Comment by MikeLing** [comment]
Hi @karlnapf , I just fix the existing comments. Maybe we could add the computational complexity later?

**Comment by karlnapf** [comment]
Ok lets do complexity later

**Comment by karlnapf** [comment]
Use git amend to correct commits, now you need to squash them (we want only one)

File: src/shogun/multiclass/KDTreeKNNSolver.h [hunk 14]
```diff
@@ -16,10 +16,9 @@ namespace shogun
 {
 
 /**
- * The KDTREEKNNSolver class is a solver class which inherit from KNNSolver class.
- * It use a k-d tree (short for k-dimensional tree) to delivery the nearest neighbor searches
- * involving the multidimensional search key.
- * Check https://en.wikipedia.org/wiki/K-d_tree for more detail about K-D tree.
+ * KD-tree solver. It uses k-d tree (short for k-dimensional tree) to speed up the 
+ * nearest neighbour computation. involving the multidimensional search key.
```
Comment (karlnapf) [inline src/shogun/multiclass/KDTreeKNNSolver.h]: Sentences start with capital letter. The second sentence misses a subject. I also dont understand what that is supposed to mean

File: src/shogun/multiclass/CoverTreeKNNSolver.h [hunk 15]
```diff
@@ -17,9 +17,8 @@ namespace shogun
 {
 
 /**
- * The CoverTreeKNNSolver class is a solver class which inherit from KNNSolver class. 
- * It use the cover tree to speed up the nearest neighbor search and you can find more
- * detail information on https://en.wikipedia.org/wiki/Cover_tree
+ * Cover tree solver. It uses the cover trees to speed up the nearest neighbour computation.
```
Comment (karlnapf) [inline src/shogun/multiclass/CoverTreeKNNSolver.h]: remove "the"

File: src/shogun/multiclass/KDTreeKNNSolver.h (hunk 14 above)
Comment (karlnapf) [inline src/shogun/multiclass/KDTreeKNNSolver.h]: Maybe just remove

**Comment by MikeLing** [comment]
@karlnapf I had fix those nits. Thank you for your reviewing :)

**Pull request merged**